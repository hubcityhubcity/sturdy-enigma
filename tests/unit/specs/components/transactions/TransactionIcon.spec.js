import { shallowMount } from "@vue/test-utils"
import TransactionIcon from "src/components/transactions/TransactionIcon"

describe(`TransactionIcon`, () => {
  let wrapper

  const transactions = [
    {
      type: `Sent`,
      group: `banking`
    },
    {
      type: `Received`,
      group: `banking`
    },
    {
      type: `Staked`,
      group: `staking`
    },
    {
      type: `Restaked`,
      group: `staking`
    },
    {
      type: `Unstaked`,
      group: `staking`
    },
    {
      type: `Claimed`,
      group: `distribution`
    },
    {
      type: `Update`,
      group: `distribution`
    },
    {
      type: `Voted`,
      group: `governance`
    },
    {
      type: `Deposit`,
      group: `governance`
    },
    {
      type: `Submitted`,
      group: `governance`
    },
    {
      type: `Create`,
      group: `staking`
    },
    {
      type: `Edit`,
      group: `staking`
    },
    {
      type: `Unjail`,
      group: `staking`
    }
  ]

  for (let i = 0; i < transactions.length; i++) {
    it(`renders a ${transactions[i].type} transaction icon`, () => {
      wrapper = shallowMount(TransactionIcon, {
        propsData: {
          transactionGroup: transactions[i].group,
          transactionType: transactions[i].type
        }
      })
      expect(wrapper.element).toMatchSnapshot()
      expect(wrapper.find(".tx__icon img").classes()).toContain(
        transactions[i].group
      )
      expect(wrapper.find(".tx__icon img").attributes("src")).toEqual(
        `/img/icons/activity/${transactions[i].type}.svg`
      )
    })
  }
})
