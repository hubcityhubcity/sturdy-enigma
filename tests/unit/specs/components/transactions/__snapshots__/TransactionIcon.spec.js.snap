// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionIcon renders a Claimed transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Claimed"
    class="distribution"
    src="/img/icons/activity/Claimed.svg"
  />
</div>
`;

exports[`TransactionIcon renders a Create transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Create"
    class="staking"
    src="/img/icons/activity/Create.svg"
  />
</div>
`;

exports[`TransactionIcon renders a Deposit transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Deposit"
    class="governance"
    src="/img/icons/activity/Deposit.svg"
  />
</div>
`;

exports[`TransactionIcon renders a Edit transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Edit"
    class="staking"
    src="/img/icons/activity/Edit.svg"
  />
</div>
`;

exports[`TransactionIcon renders a Received transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Received"
    class="banking"
    src="/img/icons/activity/Received.svg"
  />
</div>
`;

exports[`TransactionIcon renders a Restaked transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Restaked"
    class="staking"
    src="/img/icons/activity/Restaked.svg"
  />
</div>
`;

exports[`TransactionIcon renders a Sent transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Sent"
    class="banking"
    src="/img/icons/activity/Sent.svg"
  />
</div>
`;

exports[`TransactionIcon renders a Staked transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Staked"
    class="staking"
    src="/img/icons/activity/Staked.svg"
  />
</div>
`;

exports[`TransactionIcon renders a Submitted transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Submitted"
    class="governance"
    src="/img/icons/activity/Submitted.svg"
  />
</div>
`;

exports[`TransactionIcon renders a Unjail transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Unjail"
    class="staking"
    src="/img/icons/activity/Unjail.svg"
  />
</div>
`;

exports[`TransactionIcon renders a Unstaked transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Unstaked"
    class="staking"
    src="/img/icons/activity/Unstaked.svg"
  />
</div>
`;

exports[`TransactionIcon renders a Update transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Update"
    class="distribution"
    src="/img/icons/activity/Update.svg"
  />
</div>
`;

exports[`TransactionIcon renders a Voted transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Voted"
    class="governance"
    src="/img/icons/activity/Voted.svg"
  />
</div>
`;
